src/backend/utils/description.ts

```diff
@@ -21,64 +21,73 @@ export abstract class Description {
+        case 'NgClass':
+          return p(Description._getClassDesc(instance));
+        case 'NgStyle':
+          return p(Description._getNgClassDesc(instance));
+        case 'NgFormModel':
+          return p(Description._getNgFormModelDesc(instance));
+        case 'NgFormControl':
+          return p(Description._getFormControlDesc(instance));
+        case 'NgControlStatus':
+          return p(Description._getControlStatusDesc(instance));
+        case 'NgModel':
+          return p(Description._getNgModelDesc(instance));
+        case 'NgForm':
+          return p(Description._getNgFormDesc(instance));
+        default:
+          return p([]);
+      }
+    } catch (e) {
+      return p([]);
     }
-    return p([]);
   }
 
   public static getComponentDescription(debugElement: any): Array<Property> {
     if (debugElement == null) {
       return [];
     }
 
-    const element: any = debugElement.nativeElement;
-
-    const componentName = debugElement.componentInstance
-      ? debugElement.componentInstance.constructor.name
-      : element.tagName.toLowerCase();
-
-    switch (componentName) {
-      case 'a':
-        return [
-          { key: 'text', value: element.text },
-          { key: 'url', value: element.hash }
-        ];
-      case 'NgSelectOption':
-        return Description._getSelectOptionDesc(element);
-      case 'NgIf':
-        return Description._getNgIfDesc(debugElement.componentInstance);
-      case 'NgControlName':
-        return Description._getControlNameDesc
+    try {
+      const element: any = debugElement.nativeElement;
+
+      const componentName = debugElement.componentInstance
+        ? debugElement.componentInstance.constructor.name
+        : element.tagName.toLowerCase();
+
+      switch (componentName) {
+        case 'a':
+          return [
+            {key: 'text', value: element.text},
+            {key: 'url', value: element.hash}
+          ];
+        case 'NgSelectOption':
+          return Description._getSelectOptionDesc(element);
+        case 'NgIf':
+          return Description._getNgIfDesc(debugElement.componentInstance);
+        case 'NgControlName':
+          return Description._getControlNameDesc
           (debugElement.componentInstance);
-      case 'NgSwitch':
-        return Description._getNgSwitchDesc(debugElement.componentInstance);
-      case 'NgSwitchWhen':
-        return Description._getNgSwitchWhenDesc
+        case 'NgSwitch':
+          return Description._getNgSwitchDesc(debugElement.componentInstance);
+        case 'NgSwitchWhen':
+          return Description._getNgSwitchWhenDesc
           (debugElement.componentInstance);
-      case 'NgSwitchDefault':
-        return Description._getNgSwitchWhenDesc
+        case 'NgSwitchDefault':
+          return Description._getNgSwitchWhenDesc
           (debugElement.componentInstance);
+        default:
+          return [];
+      }
+    } catch (e) {
+      return [];
```

Comment:
No, this isn't the case, the description will never have more than one property, hence why each switch case returns immediately once an element type is found and the corresponding property has been read. Only one property read is required for building out the description info (hence the switch statement to determine which property read to perform). If the specific property read fails then the element will have no description properties (empty array), which I think makes sense.